api/src/main/java/org/hawkular/apm/api/model/events/EndpointRef.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.apm.api.model.events;
+
+import java.io.Externalizable;
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+
+import org.hawkular.apm.api.model.Constants;
+import org.hawkular.apm.api.utils.EndpointUtil;
+import org.hawkular.apm.api.utils.SerializationUtil;
+
+/**
+ * This class represents an endpoint reference.
+ *
+ * @author gbrown
+ */
+public class EndpointRef implements Externalizable {
```

Comment:
I've used in other places, so consistent - but probably not such a performance issue as it used to be, so I'll create a separate PR to replace all uses.